File: python/0137.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = sorted(nums)
        
        i = 0
        while i < len(nums):
            #last one
            if i == len(nums) - 1:
                return nums[i]
            # jump
            if nums[i] == nums[i + 1]:
                i += 2
            else:
                return nums[i]
            i += 1

File: cpp/0137.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> umap;

        for (int i = 0; i < nums.size(); i++) {
            if (umap.find(nums[i]) == umap.end()) {
                umap[nums[i]] = 1;
            } else {
                umap[nums[i]]++;
                if (umap[nums[i]] == 3) {
                    umap.erase(nums[i]);
                }
            }
        }

        return umap.begin()->first;
    }
};
